dace/sdfg/utils.py

```diff
@@ -1082,7 +1082,8 @@ def fuse_states(sdfg: SDFG, permissive: bool = False, progress: bool = None) ->
                 if u in skip_nodes or v in skip_nodes:
                     continue
                 candidate = {StateFusion.first_state: sd.node_id(u), StateFusion.second_state: sd.node_id(v)}
-                sf = StateFusion(sd, id, -1, candidate, 0, override=True)
+                sf = StateFusion()
+                sf.setup_match(sd, id, -1, candidate, 0, override=True)
```

Comment:
Is it still possible to initialize a transformation object with the Transformation's init method or use of `setup_match` is obligatory?

Reply:
Yes, the use of `setup_match()` is only necessary before calling `apply`